src/rendering/MaterialParser/ConfigLoader.cc

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+// This code is adapted from https://wiki.ogre3d.org/All-purpose+script+parser
```

Comment:
@iche033 https://wiki.ogre3d.org/All-purpose+script+parser has multiple versions of this script, some with stated bugs. Do you know if our version has all the bugs fixed?

Reply:
I believe we have the ogre-independent version with all the bugfixes.

Reply:
Awesome!